exchange/bidder_test.go

```diff
@@ -1231,6 +1240,100 @@ func TestSetAssetTypes(t *testing.T) {
 	}
 }
 
+func TestCallRecordAdapterConnections(t *testing.T) {
+	// Setup mock server
+	respStatus := 200
+	respBody := "{\"bid\":false}"
+	server := httptest.NewServer(mockHandler(respStatus, "getBody", respBody))
+	defer server.Close()
+
+	requestHeaders := http.Header{}
+	requestHeaders.Add("Content-Type", "application/json")
+
+	// declare requestBid parameters
+	bidAdjustment := 2.0
+	mockBidderResponse := &adapters.BidderResponse{
+		Bids: []*adapters.TypedBid{
+			{
+				Bid: &openrtb.Bid{
+					Price: 3.0,
+				},
+				BidType:      openrtb_ext.BidTypeBanner,
+				DealPriority: 4,
+			},
+		},
+	}
+
+	bidderImpl := &goodSingleBidder{
+		httpRequest: &adapters.RequestData{
+			Method:  "POST",
+			Uri:     server.URL,
+			Body:    []byte("{\"key\":\"val\"}"),
+			Headers: http.Header{},
+		},
+		bidResponse: mockBidderResponse,
+	}
+
+	// Setup mock metrics
+	metrics := &pbsmetrics.MetricsEngineMock{}
+	expectedAdapterName := openrtb_ext.BidderAppnexus
+	compareConnInfo := func(info httptrace.GotConnInfo) bool { return !info.Reused }
+	compareConnWaitTime := func(dur time.Duration) bool { return dur.Nanoseconds() > 0 }
+
+	metrics.On("RecordAdapterConnections", expectedAdapterName, mock.MatchedBy(compareConnInfo), mock.MatchedBy(compareConnWaitTime)).Once()
+
+	// Run requestBid using an http.Client with a mock handler
+	bidder := adaptBidder(bidderImpl, server.Client(), &config.Configuration{}, metrics, openrtb_ext.BidderAppnexus)
+	_, errs := bidder.requestBid(context.Background(), &openrtb.BidRequest{}, "test", bidAdjustment, currencies.NewRateConverterDefault().Rates(), &adapters.ExtraRequestInfo{})
+
+	// Assert no errors
+	assert.Equal(t, 0, len(errs), "bidder.requestBid returned errors %v \n", errs)
+
+	// Assert RecordAdapterConnections() was called with the parameters we expected
+	metrics.AssertExpectations(t)
+}
+
+func TestCallRecordRecordDNSTime(t *testing.T) {
+	//setup mock server to run
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(200)
+		if r.Method == "GET" {
+			w.Write([]byte("getBody"))
+		} else {
+			w.Write([]byte("postBody"))
+		}
+	})
+	server := httptest.NewServer(handler)
+	defer server.Close()
+
+	//setup influx metrics that give us access to the DNSLookupTimer
+	cfg := config.Configuration{}
```

Comment:
Would it be simpler to use a metrics mock here instead of relying on the Influx implementation?

Reply:
surprisingly, it was a little bit hard to make them work. Will keep looking into it tomorrow.

Reply:
What was the trouble? You should need 1 expectation on the mock RecordDNSTime with a 0 duration, yeah?

Reply:
Yep. that 0 duration value was not testing anything. Except that `ClientTrace.DNSDone()` wasn't been called.